src/server/test/completion.spec.ts

```diff
@@ -0,0 +1,226 @@
+jest.mock('ic-mops/commands/add');
+import {
+    CompletionItem,
+    CompletionList,
+    Connection,
+    InitializeResult,
+} from 'vscode-languageserver/node';
+import { URI } from 'vscode-uri';
+import { clientInitParams, setupClientServer } from '../test/mock';
+import { cwd } from 'node:process';
+import { EventEmitter } from 'node:events';
+import { wait, waitForNotification } from './helpers';
+
+const initText = `
+import A = "a";
+import B = "b";
+import Blob = "mo:base/Blob";
+`;
+
+const workText =
+    initText +
+    `
+let a =     A.
+let b =     B.
+let c =     C.
+let d =  Blob.
+let e = Array.
+`;
+
+const rootUri = URI.file(`${cwd()}/test/completion`);
+
+const file = {
+    uri: `${rootUri}/not-exist.mo`,
+    textDocument: {
+        uri: `${rootUri}/not-exist.mo`,
+        languageId: 'motoko',
+        version: 1,
+        text: initText,
+    },
+};
+
+describe('completion', () => {
+    let client: Connection;
+    let server: Connection;
+
+    beforeAll(async () => {
+        console.log(`Running test on ${rootUri}`);
+
+        const emitter = new EventEmitter();
+
+        [client, server] = setupClientServer(true);
+
+        client.onNotification('custom/initialized', () => {
+            emitter.emit('custom/initialized');
+        });
+
+        await client.sendRequest<InitializeResult>(
+            'initialize',
+            clientInitParams(rootUri),
+        );
+
+        await client.sendNotification('initialized', {});
+
+        await waitForNotification('custom/initialized', emitter);
+
+        await client.sendNotification('textDocument/didOpen', {
+            textDocument: file.textDocument,
+        });
+
+        await wait(0.5);
+
+        await client.sendNotification('textDocument/didChange', {
+            textDocument: {
+                uri: file.uri,
+                version: 2,
+            },
+            contentChanges: [
+                {
+                    text: workText,
+                },
+            ],
+        });
+
+        await wait(0.5);
+    });
+    afterAll(async () => {
+        await wait(2);
+        server.dispose();
+        client.dispose();
+    });
+    it('local module completion with import 1', async () => {
```

Comment:
Just a nitpick, but AFAIU, the philosophy of `it` is to write test names in the format "it should ..." or "it is ..." or "it expects ...", etc. So here, for example, it could be `it('provides completions for the "a" module', async () => ...)`.